### DISCUSSION TITLE
Improving DMC Charts

### DISCUSSION THREAD

**Discussion by namakshenas** [snehilvj/dash-mantine-components#430] [Ideas]
Hey folks,

I’ve been playing around with the charts in DMC recently. Just to clarify, these charts are built using the Recharts.js engine, brought into the Mantine ecosystem. Now, I have no idea why they went with Recharts.js over some other awesome JS charting libraries. Honestly, I found the DMC charting system (or more accurately, Mantine’s implementation) a bit underwhelming—it’s not super robust, has some quirks, and doesn’t offer much in terms of customization. Plus, it’s missing a bunch of cool chart types like heatmaps, boxplots, treemaps, and more. As a data analyst, those are pretty much essentials for my projects.

So, I did some digging. Turns out, a while back, some people tried integrating ApexCharts and ECharts into the Dash core. But sadly, those efforts weren’t maintained and are now riddled with bugs.

Here’s my proposal: we could go two ways. Option one, we gather a few folks from the community to build an add-on package for DMC that supports ECharts. Option two, I keep rolling with clientside_callback to meet my needs. If you’re curious, here’s a quick heatmap example I have created using clientside_callback and ECharts.js:

https://py.cafe/namakshenas/dash-echarts-heatmap

**Comment by AnnMarieW** [comment]
Hi @namakshenas 

Your e-charts looks good 🙂 

Another option is to use [figure templates](https://plotly.com/python/templates/) to style Plotly charts so they look nice with the default light and dark themes in DMC.  Then you will have all the advanced features of Plotly charts using the `dcc.Graph` component.  The only downside vs using the DMC charts is that you need to use callbacks to update the figure’s template when switching between a light and dark theme. 

I’m actually working on the figure templates now and will have a PR ready to go shortly.

**Comment by namakshenas** [comment]
Hi @AnnMarieW 

I’m not a fan of Plotly charts. I’ve been using them for a long time, and honestly, they’re a pain when it comes to graphical details, smooth animations, and those little finishing touches. Like, can you believe it took the Plotly community five whole years to add gradient support to area charts?! Crazy, right?
